### DISCUSSION TITLE
Elm support

### DISCUSSION THREAD

**Discussion by zadeviggers** [withastro/roadmap#207] [Proposal]
I don't think that it would be particularly challenging to implement, givien that there are already integrations/renderers that run special compilers, and Elm comes with a built in way to mount onto a single DOM node (although most UI franeworks have that these days).

This probably wouldn't be too time-consuming to implement (and not overly complex, so lots should be able to be done by non-core community members), so think it's feasable to have support for Elm before v1.0 comes out in late July, although it would probably be unlikely.

I just thought I'd start a discussion about this - I'm no Elm expert so I probablly don't know enough about it's internals to write a full RFC, but I'm happy to try help implement it once it's written.

**Comment by sevfurneaux** [comment]
Hi @zadeviggers 👋, here's my initial pass at adding Elm: https://github.com/sevfurneaux/astro-elm.

For rendering, it uses a forked version of [vite-plugin-elm](https://github.com/hmsk/vite-plugin-elm) which [patches the compiled output](https://github.com/sevfurneaux/astro-elm/blob/main/vite-plugin-elm-forked/index.mjs#L99-L112) exposing the `view`'s virtual dom nodes and [renders them to a HTML string](https://github.com/sevfurneaux/astro-elm/blob/main/server.js#L10-L49).

What is a final challenge, is getting Elm compiled in the client output. Currently in-browser it is outputting the `Main.elm` file as opposed to the compiled JavaScript output:

<img width="1116" alt="Screenshot 2022-09-05 at 23 21 40" src="https://user-images.githubusercontent.com/1056416/188517065-477862fe-ea3d-4d55-ae55-4e769d9b789d.png">

**Reply by zadeviggers** [reply]
Woah this is really cool! Thanks for working on this! I would love to be able to help you out with this, but unfortunately, I've got some exams coming up pretty soon, so I've got to spend most of my time studying for them.
Keep up the great work, and please keep us all updated!
